core/trino-main/src/main/java/io/trino/execution/SetColumnTypeTask.java

```diff
@@ -78,15 +77,18 @@ public ListenableFuture<Void> execute(
         QualifiedObjectName qualifiedObjectName = createQualifiedObjectName(session, statement, statement.getTableName());
         RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, qualifiedObjectName);
         if (redirectionAwareTableHandle.getTableHandle().isEmpty()) {
-            String exceptionMessage = format("Table '%s' does not exist", qualifiedObjectName);
+            String additionalInformation;
             if (metadata.getMaterializedView(session, qualifiedObjectName).isPresent()) {
-                exceptionMessage += ", but a materialized view with that name exists.";
+                additionalInformation = ", but a materialized view with that name exists.";
             }
             else if (metadata.getView(session, qualifiedObjectName).isPresent()) {
-                exceptionMessage += ", but a view with that name exists.";
+                additionalInformation = ", but a view with that name exists.";
+            }
+            else {
+                additionalInformation = "";
             }
             if (!statement.isTableExists()) {
-                throw semanticException(TABLE_NOT_FOUND, statement, exceptionMessage);
+                throw semanticException(TABLE_NOT_FOUND, statement, "Table '%s' does not exist%s", qualifiedObjectName, additionalInformation);
```

Comment:
just do
```
throw semanticException(TABLE_NOT_FOUND, statement, "%s", exceptionMessage);
```
(reverting other changes)

i think it will be more readable